File: python/monitor.py


from rules.rule import initRules
from actions.action import initActions
from parse import parseString


class Monitor:
    def __init__(self):
        self.d_rules = initRules()
        self.d_actsions = initActions()

    def parse(self, line):
        event = parseString(line)
        if not event:
            throw Exception("parse string failed")
            
        for r in self.d_rules:
            s = r.check(event)
            if not s:
                continue
            for act in self.d_actions:
                act.tackAction(s)

File: golang/src/app/monitor.go
package main

import (
    rl "rules"
    ac "actions"
)
type Monitor struct {
    d_rules []rl.Rule
    d_actions []ac.Action
}

func (m*Monitor)Init(){
    m.d_actions = ac.InitAllActions()
    m.d_rules = rl.InitAllRules()
}

func (m*Monitor)MparseLine(line string)string{
    event,rb := ParseLine(line)
    if rb == false {
        return "Parse line failed"
    }
    for _, rule := range m.d_rules {
        ts := rule.CheckEvent(&event)
        if len(ts) == 0 {
            continue
        }
        for _,act := range m.d_actions {
            act.TakeAction(ts)
        }
    }
    return ""
}

